mutiny-core/src/ldkstorage.rs

```diff
@@ -262,6 +266,64 @@ impl MutinyNodePersister {
             })
             .collect())
     }
+
+    /// Persists the failed spendable outputs to storage.
+    /// Previously failed spendable outputs are not overwritten.
+    ///
+    /// This is used to retry spending them later.
+    pub fn persist_failed_spendable_outputs(
+        &self,
+        failed: Vec<SpendableOutputDescriptor>,
+    ) -> anyhow::Result<()> {
+        let key = self.get_key(FAILED_SPENDABLE_OUTPUT_DESCRIPTOR_KEY);
+
+        // get the currently stored descriptors encoded as hex
+        // if there are none, use an empty vector
+        let mut descriptors: Vec<String> = self.storage.get(&key)?.unwrap_or_default();
```

Comment:
If there is a storage reading problem but there are outputs in there, won't that overwrite? Shouldn't it only trigger this if it's `None`, but not a key reading error?

Reply:
If it errors or is `None` it will return an empty vector and the retry logic is skipped.

Reply:
But this is the persist logic part where it could be overwritten with an empty vec + the new descriptor

Reply:
this is calling `get_failed_spendable_outputs`

Reply:
it calls `get` on `storage` and that returns `Result<Option<T>, MutinyError>` and so `unwrap_or_default` will execute the `unwrap` case on both `MutinyError` and `None`, right?

Reply:
I do `self.storage.get(&key)?.unwrap_or_default();` so the `?` will propgate the error up and fail this function, the `unwrap_or_default` will turn create an empty vec if it is None. So if there is any errors it will be properly handled, the `unwrap_or_default` is only handling if there is no stored failed spendable outputs

Reply:
Ahh, I see. Yeah that makes sense thansk for clarifying.